core/src/main/scala/kafka/server/FetchSession.scala

```diff
@@ -71,6 +75,7 @@ object FetchSession {
   * localLogStartOffset is the log start offset of the partition on this broker.
   */
 class CachedPartition(val topic: String,
+                      var topicId: Uuid,
```

Comment:
I thought about this, but I was worried about some weirdness where we need to support partitions before and after they have an id. This may also cause problems in cases where IDs may change. Consider the code ` val cachedPart = session.partitionMap.find(new CachedPartition(topicPart))`  This is used in the path of deleting partitions with stale IDs. We would need to know the topic ID to find the partition here.  I could see potential issues where we no longer have the ID and would have trouble removing it.